### PR TITLE
🌱 Remove RequeueAfterError from machine pool controller

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: This PR cleans the use of RequeueAfterError from the MachinePool controller.

Depends on #3445 

/hold

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #3370

### CODE REVIEW

File: exp/controllers/machinepool_controller_phases.go [hunk 1]
```diff
@@ -154,110 +156,111 @@ func (r *MachinePoolReconciler) reconcileExternal(ctx context.Context, cluster *
 }
 
 // reconcileBootstrap reconciles the Spec.Bootstrap.ConfigRef object on a MachinePool.
-func (r *MachinePoolReconciler) reconcileBootstrap(ctx context.Context, cluster *clusterv1.Cluster, m *expv1.MachinePool) error {
+func (r *MachinePoolReconciler) reconcileBootstrap(ctx context.Context, cluster *clusterv1.Cluster, m *expv1.MachinePool) (ctrl.Result, error) {
+	logger := r.Log.WithValues("machinepool", m.Name, "namespace", m.Namespace)
```
Comment (detiber) [inline exp/controllers/machinepool_controller_phases.go:160]:
```suggestion
	logger := r.Log.WithValues("cluster", cluster.Name, "machinepool", m.Name, "namespace", m.Namespace)
```

File: exp/controllers/machinepool_controller_phases.go [hunk 2]
```diff
@@ -154,110 +156,111 @@ func (r *MachinePoolReconciler) reconcileExternal(ctx context.Context, cluster *
...
 	m.Spec.Template.Spec.Bootstrap.DataSecretName = pointer.StringPtr(secretName)
 	m.Status.BootstrapReady = true
-	return nil
+	return ctrl.Result{}, nil
 }
 
 // reconcileInfrastructure reconciles the Spec.InfrastructureRef object on a MachinePool.
-func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, cluster *clusterv1.Cluster, mp *expv1.MachinePool) error {
+func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, cluster *clusterv1.Cluster, mp *expv1.MachinePool) (ctrl.Result, error) {
+	logger := r.Log.WithValues("machinepool", mp.Name, "namespace", mp.Namespace)
```
Comment (detiber) [inline exp/controllers/machinepool_controller_phases.go:211]:
```suggestion
	logger := r.Log.WithValues("cluster", cluster.Name, "machinepool", mp.Name, "namespace", mp.Namespace)
```

File: exp/controllers/machinepool_controller_noderef.go [hunk 3]
```diff
@@ -43,11 +44,11 @@ type getNodeReferencesResult struct {
 	ready      int
 }
 
-func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, cluster *clusterv1.Cluster, mp *expv1.MachinePool) error {
+func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, cluster *clusterv1.Cluster, mp *expv1.MachinePool) (ctrl.Result, error) {
 	logger := r.Log.WithValues("machinepool", mp.Name, "namespace", mp.Namespace)
```
Comment (detiber) [inline exp/controllers/machinepool_controller_noderef.go:48]:
```suggestion
	logger := r.Log.WithValues("cluster", cluster.Name, "machinepool", mp.Name, "namespace", mp.Namespace)
```

**Comment by CecileRobertMichon** [comment]
manual test:

```
kubectl --kubeconfig=./kubeconfig get nodes
NAME                                      STATUS   ROLES    AGE     VERSION
capz-vmss-requeue-2-control-plane-8fprq   Ready    master   8m6s    v1.18.8
capz-vmss-requeue-2-control-plane-9rn4s   Ready    master   2m22s   v1.18.8
capz-vmss-requeue-2-control-plane-pb9w4   Ready    master   5m52s   v1.18.8
capz-vmss-requeue-2-mp-0000002            Ready    <none>   6m4s    v1.18.8
capz-vmss-requeue-2-mp-0000003            Ready    <none>   6m1s    v1.18.8
```

**Pull request merged**